components/Table/index.jsx

```diff
@@ -1,91 +1,202 @@
 
 import styles from './index.scss';
 
-function getRows (cols) {
-  return (
-    [...Array(cols[0].length)].reduce((acc, item, index) => {
-      acc[index] = [];
-      cols.forEach(col => {
-        acc[index].push(col[index]);
+class Table extends Component {
+  static propTypes = {
+    columns: PropTypes.arrayOf(PropTypes.shape({
+      // String that matches to data key.
+      accessor: PropTypes.string.isRequired,
+      // For custom cells. You can use either
+      // an element or custom func which will
+      // receive a value param.
+      // ie: value => <span>{value}</span>
+      cell: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.func
+      ]),
+      // Header is the value that will be shown
+      // in the hr cell per column
+      header: PropTypes.oneOfType([
+        PropTypes.number,
+        PropTypes.element,
+        PropTypes.bool,
+        PropTypes.string
+      ]),
+      sortable: PropTypes.bool
+    })).isRequired,
+    dataRows: PropTypes.arrayOf(PropTypes.object),
+    initialSortingColumn: PropTypes.number
+  };
+
+  static defaultProps = {
+    dataRows: [],
+    initialSortingColumn: -1
+  };
+
+  state = {rows: this.props.dataRows} // eslint-disable-line
+
+  componentDidMount () {
+    const {initialSortingColumn, columns} = this.props;
+
+    if (initialSortingColumn >= 0) {
+      this.handleSorting(columns[initialSortingColumn].accessor, 1);
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const {dataRows, initialSortingColumn, columns} = this.props;
+
+    if (JSON.stringify(nextProps.dataRows) !== JSON.stringify(dataRows)) {
+      this.setState({rows: nextProps.dataRows}, () => {
+        if (initialSortingColumn >= 0) {
+          this.handleSorting(columns[initialSortingColumn].accessor, 1);
+        }
       });
-      return acc;
-    }, [])
-  );
-}
+    }
+  }
 
-function getRow (value, index) {
-  if (!value) return <td key={index} />;
-  return (
-    <td key={index}>
-      {value}
-    </td>);
-}
+  /**
+   * Sort the rows based on
+   * column and sort order
+   */
+  handleSorting = (accessor, sortOrder) => {
+    const {rows} = this.state;
 
-function getTitles (value, index) {
-  return !value ?
-    <th key={index} className={styles.emptyHeader} /> : // eslint-disable-line react/no-array-index-key
-    <th key={index}><span>{value}</span></th>; // eslint-disable-line react/no-array-index-key
-}
+    rows.sort((a, b) => {
+      // If a data entry is undefined it must be set it to
+      // an empty string in order to make a correct
+      // comparison
+      const aVal = a[accessor] || '';
+      const bVal = b[accessor] || '';
+      const isString = typeof aVal === 'string' && Number.isNaN(parseFloat(aVal));
+
+      // If row data contains numbers
+      // we should filter based on value
+      if (!isString) return ((parseFloat(aVal) || Number.NEGATIVE_INFINITY) - (parseFloat(bVal) || Number.NEGATIVE_INFINITY)) * sortOrder;
```

Comment:
Before parsing you need to replace `,` for `.`, otherwise ordering in dach is not ok, the number is truncated to int.